nearcore/src/config.rs

```diff
@@ -1330,39 +1338,79 @@ pub fn load_config(
     dir: &Path,
     genesis_validation: GenesisValidationMode,
 ) -> anyhow::Result<NearConfig> {
-    let config = Config::from_file(&dir.join(CONFIG_FILENAME))?;
-    let genesis_file = dir.join(&config.genesis_file);
+    let genesis_validation = match genesis_validation {
+        GenesisValidationMode::Full => GenesisValidationMode::Full,
+        GenesisValidationMode::UnsafeFast => GenesisValidationMode::UnsafeFast,
+    };
+    let mut validation_errors = ValidationErrors::new();
+
+    // if config.json has file issues, the program will directly panic
+    let config = Config::from_file_skip_validation(&dir.join(CONFIG_FILENAME))?;
+    // do config.json validation separately so that genesis_file, validator_file and genesis_file can be validated before program panic
+    match config.validate() {
```

Comment:
Would this work?

```suggestion
    config.validate().map_err(|e|validation_errors.push_errors(e));
```

Reply:
This would complain the Err arm not handled, I either have `let _ = `, which is kinda ugly, or the following:
`config.validate().map_or_else(
        |e|validation_errors.push_errors(e),
        |_| (),
    );`